src/crypto/algorithms/megolm.js

```diff
@@ -758,15 +758,15 @@ MegolmDecryption.prototype.decryptEvent = async function(event) {
     } catch (e) {
         let errorCode = "OLM_DECRYPT_GROUP_MESSAGE_ERROR";
 
-        if (e.message === 'OLM.UNKNOWN_MESSAGE_INDEX') {
+        if (e && e.message && e.message === 'OLM.UNKNOWN_MESSAGE_INDEX') {
```

Comment:
I think only `e && e.message === 'OLM.UNKNOWN_MESSAGE_INDEX'` is required, since if `e.message` is falsy, then it won't equal to `OLM.UNKNOWN_MESSAGE_INDEX`.